app/routes/forum/components/ThreadEditor.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Button,
+  ConfirmModal,
+  Flex,
+  Icon,
+  LoadingIndicator,
+} from '@webkom/lego-bricks';
+import { usePreparedEffect } from '@webkom/react-prepare';
+import { Field } from 'react-final-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  createThread,
+  deleteThread,
+  editThread,
+  fetchThread,
+} from 'app/actions/ForumActions';
+import { Content } from 'app/components/Content';
+import { TextInput, Form, TextArea, LegoFinalForm } from 'app/components/Form';
+import { SubmitButton } from 'app/components/Form/SubmitButton';
+import { selectThreadsById } from 'app/reducers/threads';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
+import { guardLogin } from 'app/utils/replaceUnlessLoggedIn';
+import type {
+  CreateThread,
+  DetailedThread,
+  UpdateThread,
+} from 'app/store/models/Forum';
+
+const ThreadEditor = () => {
+  const { threadId } = useParams<{ threadId?: string }>();
+  const { forumId } = useParams<{ forumId?: string }>();
+  usePreparedEffect(
+    'fetchThreadForEditor',
+    () => threadId && dispatch(fetchThread(parseInt(threadId))),
+    [threadId]
+  );
+
+  const isNew = !threadId; // Determine if it's a new forum based on the absence of forumId
```

Comment:
```suggestion
  const isNew = !threadId; // Determine if it's a new forum based on the absence of threadId
```

Kind of an excessive comment though - does it not go without saying?